scan.go

```diff
@@ -261,7 +262,11 @@ func Scan(rows Rows, db *DB, mode ScanMode) {
 						}
 					}
 				} else {
-					elem = reflect.New(reflectValueType)
+					if isPtr {
```

Comment:
We can change this to `if isPtr && db.RowsAffected > 0`? 

Seems not necessary to create a new struct for the first record.

Reply:
@jinzhu do you mean `if isPtr && db.RowsAffected == 0`?

Reply:
Oh no, you are correct, making this change now.